Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/Cancel.hs

```diff
@@ -103,6 +107,18 @@ cancel bookingId _ req = do
           Right res' -> do
             let merchantOperatingCityId = booking.merchantOperatingCityId
             disToPickup <- driverDistanceToPickup booking.merchantId merchantOperatingCityId (getCoordinates res'.currPoint) (getCoordinates booking.fromLocation)
+            now <- getCurrentTime
+            let driverBookingDuration = roundToIntegral $ diffUTCTime now booking.createdAt
+            distanceRes <-
+              Maps.getDistance booking.merchantId merchantOperatingCityId $
```

Comment:
Why to use Google API, we can use `OSRM` API, right?

Reply:
Yes this also need @sidsethu 's views . Discussion pending with him